lib/actions/apiV2.js

```diff
@@ -706,12 +756,255 @@ export const findPatternsForRoute = (params) =>
+  alerts: aggregateAlerts(
+    leg.agency?.alerts,
+    leg.route?.alerts,
+    leg.to?.stop?.alerts,
+    leg.from?.stop?.alerts
+  ),
+  alightRule: pickupDropoffTypeToOtp1(leg.dropoffType),
+  boardRule: pickupDropoffTypeToOtp1(leg.pickupType),
+  dropOffBookingInfo: {
+    latestBookingTime: leg.dropOffBookingInfo
+  },
+  from: {
+    ...leg.from,
+    stopCode: leg.from.stop?.code,
+    stopId: leg.from.stop?.gtfsId
+  },
+  route: leg.route?.shortName,
+  routeColor: leg.route?.color,
+  routeId: leg.route?.id,
+  routeLongName: leg.route?.longName,
+  routeShortName: leg.route?.shortName,
+  routeTextColor: leg.route?.textColor,
+  to: {
+    ...leg.to,
+    stopCode: leg.to.stop?.code,
+    stopId: leg.to.stop?.gtfsId
+  },
+  tripHeadsign: leg.trip?.tripHeadsign,
+  tripId: leg.trip?.gtfsId
+})
+
+const queryParamConfig = { modeButtons: DelimitedArrayParam }
+
+export function routingQuery(searchId = null, updateSearchInReducer) {
+  return function (dispatch, getState) {
+    const state = getState()
+    const { config, currentQuery, modeSettingDefinitions } = state.otp
+    const persistenceMode = getPersistenceMode(config.persistence)
+    const activeItinerary = getActiveItinerary(state)
+
+    const isNewSearch = !searchId
+    if (isNewSearch) searchId = randId()
+    // Don't permit a routing query if the query is invalid
+    if (!queryIsValid(state)) {
+      console.warn('Query is invalid. Aborting routing query', currentQuery)
+      return
+    }
+
+    const {
+      bannedTrips,
+      date,
+      departArrive,
+      modes,
+      numItineraries,
+      routingType,
+      time
+    } = currentQuery
+    const arriveBy = departArrive === 'ARRIVE'
+
+    // Retrieve active mode keys from URL parameters or configuration defaults
+    const urlSearchParams = new URLSearchParams(getUrlParams())
+    const activeModeKeys =
+      decodeQueryParams(queryParamConfig, {
+        modeButtons: urlSearchParams.get('modeButtons')
+      }).modeButtons ||
+      config?.modes?.initialState?.enabledModeButtons ||
+      {}
+
+    // Filter mode definitions based on active mode keys
+    const activeModeButtons = filterModeDefitionsByKey(
+      config.modes?.modeButtons || [],
+      activeModeKeys
+    )
+    const activeModes = aggregateModes(activeModeButtons)
+
+    // Get mode setting values from the url, or initial state config, or default value in definition
+    const modeSettingValues = modeSettingDefinitions?.reduce((acc, setting) => {
+      const fromUrl = urlSearchParams.get(setting.key)
+      acc[setting.key] = fromUrl
+        ? convertModeSettingValue(setting, fromUrl)
+        : config?.modes?.initialState?.modeSettingValues?.[setting.key] ||
+          setting.default
+      return acc
+    }, {})
+    const modeSettings = modeSettingDefinitions?.map(
+      populateSettingWithValue(modeSettingValues)
+    )
+
+    // Get the raw query param strings to save for the rider's search history
```

Comment:
Some params aren't being parsed correctly. For instance, removing the bus option still results in itineraries with a bus leg being returned (see screenshot).

<img width="374" alt="image" src="https://github.com/opentripplanner/otp-react-redux/assets/56846598/10e369d8-7a37-467b-9c62-cfb51d54bbfb">

It looks like boolean params are converted to strings in the OTP-UI package in use, which probably resolve to `true` (see debugger screenshot).
<img width="226" alt="image" src="https://github.com/opentripplanner/otp-react-redux/assets/56846598/a5eafcc5-5b22-4da0-9f53-61437f63348a">

Reply:
Yep, great catch. That was happening because I added a new type, "SUBMODE" which behaves just like "CHECKBOX", but I forgot to add that to the utility function which converts the true/false string into an actual boolean, so it was staying as a string.

Reply:
Fixed in OTP UI, new alpha package version added to this PR.